scripts/tasks.py

```diff
@@ -509,3 +511,118 @@ def timetool_correct(config):
+
+            # Write the config file
+            config_file_name = f"{setup.exp}_sample_{i+1}.yaml"
+            config_file_path = os.path.join(subdir_path, config_file_name)
+
+            with open(config_file_path, 'w') as file:
+                yaml.dump(config_temp, file)
+            
+        logger.info('Done!')
+    else:
+        logger.info('The number of config files to generate was not defined!')
+
+def bo_aggregate_init_samples(config):
+    setup = config.setup
+    task = config.bayesian_opt
+    """ Aggregates the scores and parameters of the initial samples of the Bayesian optimization. """
+    n_samples_init = task.n_samples_init
+    # Get the names of the parameters
+    params_ranges_keys = [key for key in config if key.startswith("range_")]
+    params_names = [key.replace("range_", "") for key in params_ranges_keys]
+    n_params = len(params_names)
+    # Get the first task of the loop sequence
+    task_to_optimize = config.get(task.task_to_optimize)
+    # Get the task generating the scores
+    score_task = config.get(task.score_task)
+
+    # Get the score and the parameters of each sample
+    samples_scores = np.empty(shape=(n_samples_init, 1))
+    samples_params = np.empty(shape=(n_samples_init, n_params))
+
+    logger.info('Aggregating the scores and parameters of the initial samples.')
+    subdir_name = setup.exp + "_init_samples_configs"
+    subdir_path = os.path.join(setup.root_dir, "bayesian_opt", task.tag, subdir_name)
+    for i in range(n_samples_init):
+        # Get the parameters in the config file
+        config_file_name = f"{setup.exp}_sample_{i+1}.yaml"
+        config_file_path = os.path.join(subdir_path, config_file_name)
+        with open(config_file_path, 'r') as file:
+            config_data = yaml.safe_load(file)
+        for j in range(n_params):
+            samples_params[i,j] = config_data[task_to_optimize][params_names[j]]
+
+        # Get the score in the score output file
+        score_file_name = f"{task.tag}_sample_{i+1}_{task.fom}_n1.dat" 
+        score_file_path = os.path.join(setup.root_dir, task.score_task, score_task.tag, "hkl", score_file_name)
+        with open(score_file_path, 'r') as file:
+            lines = file.readlines()
+        data = lines[1].strip().split(',')
+        score = data[1] # The score is located at the 2nd column
+        samples_scores[i,1] = score
+
+    # Write the scores and the parameters of the samples in a .dat file
+    output_file_path = os.path.join(setup.root_dir, "bayesian_opt", task.tag, f"{setup.exp}_bayesian_opt.dat")
+    with open(output_file_path, 'w', newline='') as file:
+        writer = csv.writer(file, delimiter=',')
+        # Write a header row with column names
+        writer.writerow(['score'] + params_names)
+        # Write the data
+        data_to_write = [(score, *params) for score, params in zip(samples_scores, samples_params)]
+        writer.writerows(data_to_write)
```

Comment:
Likewise, this code is also present in the `BayesianOptimization` class - is there a reason those methods cannot be used here?

Reply:
Always slightly different, so utils function would have to be separate.